### PR TITLE
MAINTAINERS+dts: Create & Update Renesas ranges area

### PR DESCRIPTION
**Pull request by aaillet**
This PR:

- Move `rzt2m.dtsi` from `dts/arm/renesas/` to `dts/arm/renesas/rz/` to follow what has been done for other Renesas ranges.
- Create `Renesas RA` and `Renesas RZ` areas to maintain as ranges maintainers/collaborators are not the same.
- Move `rzt2m.dtsi` to `Renesas RZ` area as `Renesas R-Car` maintainers are not the good ones to request reviews/assign.

For the two new areas to maintain, I currently set them as `status: odd fixes` but if we find users to maintain/collaborate on these areas, I would be thrilled to edit this PR.
Maybe some users that work on these topics would be interested:
- RA: @soburi (worked on #60760)
- RZ: @wsipak @rdob-ant @jmichalski-ant (worked on #64566), @tgorochowik and @jbylicki (worked on #65674)

A user that would want this role should go through this process first: https://github.com/zephyrproject-rtos/zephyr/issues/new?assignees=&labels=Role+Nomination&projects=&template=006_nomination.md&title=

I will keep this PR as DNM waiting for potential maintainers to answer.

### CODE REVIEW

**Comment by lorc** [comment]
There are lots of work on RZA2M going in EPAM branches right now (https://github.com/xen-troops/zephyr/commits/rza2m_dev/), so I am really curious about @klogg opinion: should someone from EPAM side try to apply RZA collaborator/maintainer?

**Comment by soburi** [comment]
Hi, @aaillet 

> A user that would want this role should go through this process first: 

I issued #67642 for nominating myself as contributer.
I would like to take on that role about Renesas RA.

**Comment by aaillet** [comment]
> Hi, @aaillet
> 
> > A user that would want this role should go through this process first:
> 
> I issued #67642 for nominating myself as contributer. I would like to take on that role about Renesas RA.

Hi @soburi , thanks for your interest.
I don't think that you need any nomination to be a contributor (contributors are not listed in `MAINTAINERS.yml` either), see : https://docs.zephyrproject.org/latest/project/project_roles.html

You can be nominated at either Collaborator or Maintainer for a certain area, would you like one of these roles ?

@nashif is it possible for an area to be "maintained" with only collaborators and no maintainer ?

**Comment by xakep-amatop** [comment]
> There are lots of work on RZA2M going in EPAM branches right now (https://github.com/xen-troops/zephyr/commits/rza2m_dev/), so I am really curious about @klogg opinion: should someone from EPAM side try to apply RZA collaborator/maintainer?

I can be a collaborator if nobody is against it.

**Comment by aaillet** [comment]
> There are lots of work on RZA2M going in EPAM branches right now (https://github.com/xen-troops/zephyr/commits/rza2m_dev/), so I am really curious about @klogg opinion: should someone from EPAM side try to apply RZA collaborator/maintainer?

Nice to know, don't hesitate to talk about it with contributors that worked on current support too.

In order to be nominated to maintain an area, I think you have to be able to show upstreamed work about the concerned area. If you didn't upstreamed any of your work on RZA2M yet, you may not be nominated yet (I'm not sure of how nomination process work).

**Comment by klogg** [comment]
@aaillet @lorc @xakep-amatop
Indeed we did not upstream our work yet and we are not sure when this will happen, so I think it make sense to hold for now with nominations from EPAM side. Hope to clarify this soon

**Comment by soburi** [comment]
> Hi @soburi , thanks for your interest.
> I don't think that you need any nomination to be a contributor (contributors are not listed in MAINTAINERS.yml either), see : https://docs.zephyrproject.org/latest/project/project_roles.html
> 
> You can be nominated at either Collaborator or Maintainer for a certain area, would you like one of these roles ?

Since it's something I created myself, I'm running for the role of maintainer. I used collaborator on rpi pico, so I understand the general feeling.
There are not many RA reviewers, so I think it would be good for me to take on the maintainer role at this time. We will consider this again when the number of developers increases.

**Comment by aaillet** [comment]
Added @soburi as Renesas RA platforms maintainer.
His nomination issues, #67642, as been validated.

**Comment by aaillet** [comment]
> Feel free to add me as maintainer for the RZ family. Antmicro added the only board from that family and I can coordinate the efforts for the foreseeable future.

Thanks @tgorochowik , should I add any of your colleagues as collaborator for the RZ platform as well ?

**Comment by tgorochowik** [comment]
Please don't do it for now, they might have limited time and might not be very responsive while pinged. If necessary, we'll propose someone additional separately

**Review by tgorochowik — DISMISSED** [review]

**Comment by aaillet** [comment]
PR is now ready for review, can you please take a look @MaureenHelm @nashif ?

File: MAINTAINERS.yml [hunk 1]
```diff
@@ -3278,14 +3278,37 @@ Renesas SmartBond Platforms:
...
+Renesas RA Platforms:
+  status: maintained
+  maintainers:
+    - soburi
   files:
-    - soc/arm/renesas_ra/
+    - boards/arm/arduino_uno_r4/
+    - drivers/*/*ra*
     - dts/arm/renesas/ra/
+    - dts/bindings/*/*ra*
```
Comment (MaureenHelm) [inline MAINTAINERS.yml:3289]: This matches a lot more than just the Renesas RA platform drivers and bindings.
```console
$ ls dts/bindings/*/*ra*
dts/bindings/adc/raspberrypi,pico-adc.yaml
dts/bindings/clock/raspberrypi,pico-clock-controller.yaml
dts/bindings/clock/raspberrypi,pico-clock.yaml
dts/bindings/clock/raspberrypi,pico-pll.yaml
dts/bindings/clock/raspberrypi,pico-rosc.yaml
dts/bindings/clock/renesas,ra-clock-generation-circuit.yaml
dts/bindings/cpu/intel,raptor-lake.yaml
dts/bindings/disk/zephyr,ram-disk.yaml
dts/bindings/display/intel,multiboot-framebuffer.yaml
dts/bindings/display/raydium,rm67162.yaml
dts/bindings/display/raydium,rm68200.yaml
dts/bindings/display/ultrachip,uc8175.yaml
dts/bindings/display/ultrachip,uc8176.yaml
dts/bindings/display/ultrachip,uc8179.yaml
dts/bindings/display/ultrachip,uc81xx-common.yaml
dts/bindings/dma/raspberrypi,pico-dma.yaml
dts/bindings/flash_controller/raspberrypi,pico-flash-controller.yaml
dts/bindings/gpio/raspberrypi,pico-gpio.yaml
dts/bindings/gpio/raspberrypi,pico-header.yaml
dts/bindings/gpio/raspberrypi-40pins-header.yaml
dts/bindings/gpio/renesas,ra-gpio.yaml
dts/bindings/i2c/raspberrypi,pico-i2c.yaml
dts/bindings/interrupt-controller/renesas,ra-interrupt-controller-unit.yaml
dts/bindings/lora/st,stm32wl-subghz-radio.yaml
dts/bindings/memory-controllers/ite,it8xxx2-bbram.yaml
dts/bindings/memory-controllers/microchip,xec-bbram.yaml
dts/bindings/memory-controllers/nuvoton,npcx-bbram.yaml
dts/bindings/memory-controllers/nxp,flexram.yaml
dts/bindings/memory-controllers/st,stm32-backup-sram.yaml
dts/bindings/memory-controllers/st,stm32-bbram.yaml
dts/bindings/memory-controllers/st,stm32-fmc-nor-psram.yaml
dts/bindings/memory-controllers/st,stm32-fmc-sdram.yaml
dts/bindings/memory-controllers/zephyr,bbram-emul.yaml
dts/bindings/misc/raspberrypi,pico-pio-device.yaml
dts/bindings/misc/raspberrypi,pico-pio.yaml
dts/bindings/misc/renesas,ra-sci.yaml
dts/bindings/modem/u-blox,sara-r4.yaml
dts/bindings/modem/u-blox,sara-r5.yaml
dts/bindings/phy/can-transceiver-gpio.yaml
dts/bindings/phy/can-transceiver.yaml
dts/bindings/pinctrl/raspberrypi,pico-pinctrl.yaml
dts/bindings/pinctrl/renesas,ra-pinctrl.yaml
dts/bindings/pwm/raspberrypi,pico-pwm.yaml
dts/bindings/regulator/raspberrypi,core-supply-regulator.yaml
dts/bindings/reset/raspberrypi,pico-reset.yaml
dts/bindings/retained_mem/zephyr,retained-ram.yaml
dts/bindings/sensor/murata,ncp15wb473.yaml
dts/bindings/sensor/nxp,kinetis-temperature.yaml
dts/bindings/sensor/raspberrrypi,pico-temp.yaml
dts/bindings/sensor/seeed,grove-temperature.yaml
dts/bindings/serial/raspberrypi,pico-uart-pio.yaml
dts/bindings/serial/raspberrypi,pico-uart.yaml
dts/bindings/serial/renesas,ra-uart-sci.yaml
dts/bindings/spi/raspberrypi,pico-spi-pio.yaml
dts/bindings/spi/raspberrypi,pico-spi.yaml
dts/bindings/sram/mmio-sram.yaml
dts/bindings/test/vnd,array-holder.yaml
dts/bindings/test/vnd,can-transceiver.yaml
dts/bindings/test/vnd,great-grandchild-bindings.yaml
dts/bindings/test/vnd,string-array-token.yaml
dts/bindings/test/vnd,string-array-unquoted.yaml
dts/bindings/timer/raspberrypi,pico-timer.yaml
dts/bindings/usb/raspberrypi,pico-usbd.yaml
dts/bindings/watchdog/raspberrypi,pico-watchdog.yaml
```

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: MAINTAINERS.yml (hunk 1 above)
Comment (aaillet) [reply MAINTAINERS.yml:3289]: Wow, I missed that, thanks for the comment, I'm working on it

**Comment by aaillet** [comment]
Waiting for #68094 renaming Renesas RA driver to be able to better delimit these drivers

File: MAINTAINERS.yml (hunk 1 above)
Comment (aaillet) [reply MAINTAINERS.yml]: Fixed, result with current patch:

```bash
> ls -1  drivers/*/*renesas_ra*
drivers/clock_control/clock_control_renesas_ra.c
drivers/clock_control/Kconfig.renesas_ra
drivers/gpio/gpio_renesas_ra.c
drivers/gpio/Kconfig.renesas_ra
drivers/interrupt_controller/intc_renesas_ra_icu.c
drivers/interrupt_controller/Kconfig.renesas_ra
drivers/pinctrl/Kconfig.renesas_ra
drivers/pinctrl/pinctrl_renesas_ra.c
drivers/serial/Kconfig.renesas_ra
drivers/serial/uart_renesas_ra.c
```

```bash
> ls -1 dts/bindings/*/*renesas,ra*
dts/bindings/clock/renesas,ra-clock-generation-circuit.yaml
dts/bindings/gpio/renesas,ra-gpio.yaml
dts/bindings/interrupt-controller/renesas,ra-interrupt-controller-unit.yaml
dts/bindings/misc/renesas,ra-sci.yaml
dts/bindings/pinctrl/renesas,ra-pinctrl.yaml
dts/bindings/serial/renesas,ra-uart-sci.yaml
```

**Review by MaureenHelm — APPROVED** [review]

**Review by tgorochowik — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**